crates/rpc/src/types/errors.rs

```diff
@@ -0,0 +1,15 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ApiError {
+    #[error("Unauthorized")]
+    Unauthorized,
+
+    #[error("{0}")]
+    NotFound(String),
+
+    #[error("Bad Request: {0}")]
+    BadRequest(String),
+}
+
+impl warp::reject::Reject for ApiError {}
```

Comment:
```suggestion
impl Reject for ApiError {}
```

could you import this